### ISSUE TITLE
Assets loader crafting invalid URLs for BitmapFont page images on Safari 14

### ISSUE THREAD

**Issue by qwertywelch** [pixijs/pixijs#8571]
### Expected Behavior
Assets loader should load BitmapFont images using valid URLs as it does on other browsers (Chrome, latest Safari, etc).

Given a pixi.js app running on https://example.com, with Assets initialized using...
```typescript
await Assets.init({ basePath: 'https://assets.example.com/' });
```
...it should load BitmapFont image files with URLs like:

`(basePath)/(image path)`
`https://assets.example.com/font-img.png`

### Current Behavior
On Mobile Safari on iOS 14 / Safari on macOS Big Sur (and probably older), dev tools report that BitmapFont page images are loaded with URLs like this:

`(origin)/(basePath minus protocol)/(image path)`
`https://example.com/assets.example.com/font-img.png`

And not being the correct path, fail to load.

### Possible Solution
I tried looking through the many path transforming parts of the relevant code but could not find any parts that isolated performed differently on the different browser versions. Though there must be something that works differently on this older Safari version; I'm guessing the "https:/" is getting stripped somehow, and treated as a relative URL. Perhaps a regex expression is getting evaluated differently?

### Steps to Reproduce
```typescript
await Assets.init({ basePath: 'https://assets.example.com/' });
```
Load a BitmapFont with the Asset loader on Safari on iOS 14, notice it fails due to incorrectly crafted URLs. Happening to me when loading bundle-referenced fonts, but I think it happens for non-bundled fonts too.

### Environment

- **`pixi.js` version**: 6.5.1
- **Browser & Version**: Safari 14
- **OS & Version**: macOS Big Sur / iOS 14 (& Older)

**Comment by qwertywelch** [comment]
Screenshot showing older Safari on top, Chrome on bottom:

<img width="735" alt="Screenshot 2022-08-09 at 2 54 11 PM" src="https://user-images.githubusercontent.com/1254417/183738898-54062774-6ef5-42aa-86fd-9068f9212861.png">


Here's a repro: https://github.com/qwertywelch/pixi-bmfont-issue-repro

**Comment by Zyie** [comment]
Hey @qwertywelch unfortunately i don't have an IOS14 device and cant reproduce on IOS15

I can point you in the right direction if you are up for trying to fix this?

[getBaseUrl](https://github.com/pixijs/pixijs/blob/ac965dd593f766a19ba9592e538a47f936f51739/packages/settings/src/adapter.ts#L39) this tries grabbing the `baseURI` then the `location.href` if that fails. MAybe IOS 14 is doing something weird here. This gets called [here](https://github.com/pixijs/pixijs/blob/ac965dd593f766a19ba9592e538a47f936f51739/packages/assets/src/utils/url/makeAbsoluteUrl.ts#L22) whenever a url is loaded

So somewhere in these two functions i believe something is going wrong

**Comment by qwertywelch** [comment]
Ok I figured it out. The code itself is running the same on both platforms, with the URL that is passed to the Loader having a slash removed from the protocol (https://example.com -> https:/example.com). Seems Chrome and latest Safari decide to still treat this as a valid absolute URL, but Safari 14 is treating it as relative. To see this, I added debug warning in Loader._getLoadPromiseAndParser to see what URL it was being given and on both platforms shows:

**Safari on iOS 14.5:**
<img width="692" alt="image" src="https://user-images.githubusercontent.com/1254417/184454485-884349b4-9b14-45cf-b80b-6c6e6999e317.png">
<img width="406" alt="image" src="https://user-images.githubusercontent.com/1254417/184454550-00dce000-aaa1-4f0a-af7f-07f5348bef43.png">

**Chrome:**
<img width="486" alt="image" src="https://user-images.githubusercontent.com/1254417/184454512-e36aa470-c23a-4668-9c3a-895b4e56e7f0.png">
<img width="390" alt="image" src="https://user-images.githubusercontent.com/1254417/184454524-e92d3e0a-80b8-4da7-8fac-8908c049f9fa.png">

Seems it is the normalize function that is doing the extra slash removal from the protocol. It appears to split the path by `/`, then passing the split array to normalizeArray which filters out the empty parts, before joining it back together. The protocol such as `https://` should be taken out before the rest of the path is passed to the normalizer. This code path of normalizing the URL appears to only be done when loading a BitmapFont's page image (loadBitmapFont.parse).

**Comment by qwertywelch** [comment]
I can confirm 6.5.2 fixes this issue on iOS 14.5, thanks @Zyie!
